### ISSUE TITLE
How do I track the current track playing in a playlist by playlist track offset?

### ISSUE THREAD

**Issue by shillshocked** [spotify/web-api#1288]
I want to be able to obtain the current track position in a playlist. So, for example, track #13 in a playlist of 50 tracks. I can obtain the track uri, but that doesn't give me the position of the track in the playlist

**Comment by 0xNF** [comment]
If you use the Playing

**Comment by 0xNF** [comment]
If you use the [Player APIs](https://developer.spotify.com/documentation/web-api/reference/player/), you can get the currently playing track, along with the context object of what's playing. The `href` of that object will be a link to a playlist the user is playing from. Then you can get the tracks of that playlist with the playlist APIs and figure out where they are in the playlist.

You can also use the [Get Current Playback](https://developer.spotify.com/documentation/web-api/reference/player/get-information-about-the-users-current-playback/) endpoint to figure out if the user has shuffle enabled (and therefore whether you can predict the next song or not).

It's a bit roundabout and requires a few upfront API calls to scan the entirety of a users playlist, but this is how I'd do it.

**Comment by shillshocked** [comment]
I'm using playlists that are in the thousands, and curating them with an app I'm developing. It would be impractical to reload the entire playlist each time I remove a track from the curation playlist. What I need to access is an index of the current track in the playlist, not the track uri. That way I can determine which will be the next and the previous, in case I want to shift up and down the playlist.

**Comment by 0xNF** [comment]
Yeah, I get what you want to do. It's just that there's no easy way to do it with the Web API.

Given that these playlists are curated, presumably by you, could you run through the playlist track APIs yourself once, keep a cache of the list, and then poll the Currently Playing endpoint and update an index pointer that you maintain yourself? Unless you anticipate the playlists being modified by uncontrollable third parties, you should be able to essentially mimic the behavior you want app-side, without any need of querying Spotify for the full playlist all the time.

**Comment by shillshocked** [comment]
But if I delete tracks in the app while tracking the current track, the index will be wrong. There's just too much room for error.

**Comment by yurigava** [comment]
> If you use the [Player APIs](https://developer.spotify.com/documentation/web-api/reference/player/), you can get the currently playing track, along with the context object of what's playing. The `href` of that object will be a link to a playlist the user is playing from. Then you can get the tracks of that playlist with the playlist APIs and figure out where they are in the playlist.
> 

I know this is an old topic, but I need to get the position of the current playing song inside a playlist. I thought of doing the way you mentioned, but what if the track is inserted multiple times? Can you think of a way to find Which is the index of the one that is playing?

**Comment by 0xNF** [comment]
There is unfortunately no good way to figure out which particular index is being played if the songs are inserted twice. I wish it weren't true, but until Spotify decides to add more detail to the API, that's definitely a thing that becomes very difficult to do reliably.

If you're up for some shenanigans, you can combine the [Get Current Playback ](https://developer.spotify.com/documentation/web-api/reference/player/get-information-about-the-users-current-playback/) with [Get Recently Played](https://developer.spotify.com/documentation/web-api/reference/player/get-recently-played/) which will give you the previous 50 tracks played (provided the user played at least 30 seconds of the track). If you combine that with [Get Playlists Tracks](https://developer.spotify.com/documentation/web-api/reference/playlists/get-playlists-tracks/), you should be able to piece together what the active index is.

It's not elegant, beautiful, or even necessarily reliable - there's plenty of room for error, but as a probabilistic patchwork, I think it should work fairly well.

**Comment by yurigava** [comment]
Thanks for the reply.
Yeah, I thought of doing that. But that would be too much work for what I need.
I chose to search the song in the playlist starting on the last position I previously got.
